components/omega/test/CMakeLists.txt

```diff
@@ -4,406 +4,133 @@
 # Data type test
 ##################
 
-set(_TestDataTypesName testDataTypes.exe)
+add_executable(testDataTypes.exe base/DataTypesTest.cpp)
 
-add_executable(${_TestDataTypesName} base/DataTypesTest.cpp)
-
-target_compile_options(
-  ${_TestDataTypesName}
-  PRIVATE
-  "-I${OMEGA_SOURCE_DIR}/src/base"
-  "-I${OMEGA_SOURCE_DIR}/src/infra"
-  ${OMEGA_CXX_FLAGS}
-)
-
-target_link_options(
-  ${_TestDataTypesName}
-  PRIVATE
-  ${OMEGA_LINK_OPTIONS}
-)
-
-target_link_libraries(${_TestDataTypesName} ${OMEGA_LIB_NAME} yakl parmetis metis)
-
-if(GKlib_FOUND)
-  target_link_libraries(${_TestDataTypesName} gklib)
-endif()
-
-add_test(
-  NAME DATA_TYPES_TEST
-  COMMAND ${MPI_EXEC} -n 1 -- ./${_TestDataTypesName}
-)
+target_link_libraries(testDataTypes.exe ${OMEGA_LIB_NAME})
 
+add_test(NAME DATA_TYPES_TEST COMMAND ${MPI_EXEC} --cpu-bind=cores -n 1 -- ./testDataTypes.exe)
```

Comment:
Completely subjective, but while at it... I would consider creating a cmake function that creates the executable, links the library, and creates the test with the proper mpi command. It could help readability maybe?

Reply:
I agree that using a CMake function is better for readability and maintenance. In fact, I need to refactor this part anyway because I think it would be better to replace the hard-coded MPI launcher flags with ones from CIME configuration XML files. Thank you!